data-model/src/main/java/org/hypertrace/core/datamodel/shared/TraceEntitiesGraph.java

```diff
@@ -87,22 +84,18 @@ private void buildParentChildRelationship(StructuredTrace trace) {
             .add(childEntity);
         childToParentEntities.computeIfAbsent(childEntity.getEntityId(), k -> new ArrayList<>())
             .add(parentEntity);
+        childrenEntityIds.add(childEntity.getEntityId());
```

Comment:
Do we need this extra field. Can't we just use childToParentEntities.keySet() in `processEntities`?

Reply:
Given that `childrenEntityIds` is not exposed in any form, i see no reason for just maintaining that extra set.

Reply:
yeah not needed, missed it. Removed